lib/html_parser.dart

```diff
@@ -4,6 +4,7 @@ import 'package:html/dom.dart' as dom;
 
 typedef CustomRender = Widget Function(dom.Node node, List<Widget> children);
 typedef OnLinkTap = void Function(String url);
+typedef ParseNodeFunction = List<Widget> Function(List<dom.Node> nodeList);
```

Comment:
Rather than creating a custom callback to parse widgets, just pass in the list of children Widgets to the Widget.

Reply:
I left it after refactoring because it is also needed to build <summary> node content, but we cannot know in advance if there is a "summary" tag, or have to determine the content of the tag outside the class, which seems to me not the best way to structure the code.